### ISSUE TITLE
Consolidate `global.googlesitekit` code into module

### ISSUE THREAD

**Issue by tofumatt** [google/site-kit-wp#2443]
## Feature Description

It's useful for us to have a global variable for 3rd-party developers, sharing references to things like `React` across modules, etc. But our way of accessing that `global.googlesitekit` variable could be improved and abstracted.

Currently we reference the `global.googlesitekit` variable directly in many files, often initialising it first if it doesn't yet exist. Eg: https://github.com/google/site-kit-wp/blob/9ccd530f9606f18e8a9f9fc819d98935a0a0c206/assets/js/element-shim.js#L27-L29 and https://github.com/google/site-kit-wp/blob/9ccd530f9606f18e8a9f9fc819d98935a0a0c206/assets/js/googlesitekit-data.js#L24-L26

It would be nice to consolidate that check + initialisation into a single place, and also to allow Site Kit code to access the "shared" global state with a new module. For instance, we could go from:

```js
if ( typeof global.googlesitekit === 'undefined' ) {
	global.googlesitekit = {};
}

if ( typeof global.googlesitekit.data === 'undefined' ) {
	global.googlesitekit.data = Data;
}
```

to:

```js
import googlesitekit from 'googlesitekit';

googlesitekit.data = googlesitekit.data || Data;
```

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* All references to `global.googlesitekit` should be replaced with an `import googlesitekit from 'googlesitekit';`.
* The `googlesitekit` module should still create and return a `global.googlesitekit` variable to be shared across modules/entrypoints.
* Initialisation of the `global.googlesitekit` should happen only once, in the `'googlesitekit'` module.
* A new entrypoint should be made for this `googlesitekit` module and it should be marked as an external in Webpack, similar to `googlesitekit-api` and `googlesitekit-data`

## Implementation Brief

* 

### Test Coverage

* 

### Visual Regression Changes

* 

## QA Brief

* 

## Changelog entry

*

**Comment by aaemnnosttv** [comment]
@tofumatt I'm not sure this issue is needed anymore after #2444. It's also somewhat similar to #2501 which is perhaps more appropriate for addressing this – what do you think?

**Comment by felixarntz** [comment]
Closing as duplicate of (superseded by) #2501
